bench/readv.ml

```diff
@@ -0,0 +1,60 @@
+(* This benchmark uses [Uring.readv] to read from /dev/zero in a loop.
+   This is intended to stress the sketch buffer. *)
+
+let buffer_size = 100   (* Use a small buffer to stress the system more *)
+let n_concurrent = 16   (* How many requests to have active at once *)
+let n_iters = 1_000_000 (* How many times to accept and resubmit *)
+
+let got = ref 0
+
+let rec wait t handle =
+  match Uring.peek t with
+  | Some { result; data = buf } -> handle result buf
+  | None ->
+    match Uring.wait t with
+    | None -> wait t handle
+    | Some { result; data = buf } -> handle result buf
+
+let run_bechmark ~polling_timeout fd =
+  got := 0;
+  (* For polling mode, [queue_depth] needs to be slightly larger than [n_concurrent] or submission
+     occasionally fails for some reason. *)
+  let t = Uring.create ?polling_timeout ~queue_depth:(n_concurrent * 2) () in
+  (* We start by submitting [n_concurrent] reads. *)
+  for _ = 1 to n_concurrent do
+    let buf = Cstruct.create buffer_size in
+    let _job : _ Uring.job = Uring.readv t fd [buf] ~file_offset:Optint.Int63.zero [buf] |> Option.get in
+    ()
+  done;
+  assert (Uring.submit t = n_concurrent);
+  (* Then we wait for reads to complete. Each time a read returns, we immediately submit another. *)
+  let t0 = Unix.gettimeofday () in
+  for _ = 1 to n_iters do
+    wait t (fun result bufs ->
+        if result < 0 then (
+          raise (Unix.Unix_error (Uring.error_of_errno result, "readv", ""))
+        ) else (
+          got := !got + result;
```

Comment:
I'd assert that `result = buffer_size`, if/when someone changes buffersize to one brazilion, it might return less.

Reply:
That wouldn't be an error, though, and we report the actual size we read so the results will still be accurate.